Python/getargs.c

```diff
@@ -848,8 +848,21 @@ convertsimple(PyObject *arg, const char **p_format, va_list *p_va, int flags,
     }
 
     case 'p': {/* boolean *p*redicate */
-        int *p = va_arg(*p_va, int *);
         int val = PyObject_IsTrue(arg);
+        if (*format == '*') {
+            // p*
+	    format++;
+            bool *p_bool = va_arg(*p_va, bool *);
+            if (val > 0)
+                *p_bool = true;
+            else if (val == 0)
+                *p_bool = false;
+            else
+                RETURN_ERR_OCCURRED;
+	    break;
+        }
```

Comment:
```suggestion
        if (*format == '*') {
            // p*
	    	format++;
            bool *p_bool = va_arg(*p_va, bool *);
          	if (val < 0) {
          		RETURN_ERR_OCCURRED;
      		}
      		*p_bool = val ? true : false;
		    break;
        }
```

For new code, we use PEP 7. I think we can even do `*p_bool = (bool)(val)`.

Reply:
> For new code, we use PEP 7.

@picnixz, I don't think it's a good idea here.  This code is a part of an old function, not a new one.  Better be consistent with the rest (e.g. use same commentary style, etc).

But code mix tabs and spaces - that should be fixed.

Reply:
There's a mix of both so I don't really know which is the best... But yes, maybe consistency here is better. However, I think we can use the alternative construction I suggested. What do you think?

Reply:
yes, ``*p_bool = (bool)(val)`` does make sense.

Reply:
Updated the suggestion, thanks Sergey.

Reply:
This still mix tabs and spaces:)

Reply:
> This still mix tabs and spaces:)

Wait, there are tabs and spaces being mixed in my suggestion? I can't see them...

Reply:
Indeed, your suggestion on https://github.com/python/cpython/pull/123097/files tab looks fixed, thanks.

Reply:
I was switching between vim and nvim, I think my configuration between the two is mismatched. Unfortunately, commiting Sergey's suggestion made this outdated, so I had to do it manually, sorry about that!